preprocessors/charts/charts.py

```diff
@@ -0,0 +1,131 @@
+# Copyright (c) 2021 IMAGE Project, Shared Reality Lab, McGill University
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# and our Additional Terms along with this program.
+# If not, see
+# <https://github.com/Shared-Reality-Lab/IMAGE-server/blob/main/LICENSE>.
+
+
+import json
+import time
+import logging
+import jsonschema
+from flask import Flask, request, jsonify
+
+from charts_utils import getLowerPointsOnLeft, getHigherPointsOnLeft
+from chart_utils import getLowerPointsOnRight, getHigherPointsOnRight
+
+app = Flask(__name__)
+
+
+@app.route('/preprocessor', methods=['POST', 'GET'])
+def get_chart_info():
+    """
+    Get the useful chart information from the given input
+    """
+    logging.debug("Received request")
+    # Load schemas
+    with open('./schemas/preprocessors/charts.schema.json') as jsonfile:
+        data_schema = json.load(jsonfile)
+    with open('./schemas/preprocessor-response.schema.json') as jsonfile:
+        schema = json.load(jsonfile)
+    with open('./schemas/definitions.json') as jsonfile:
+        definition_schema = json.load(jsonfile)
+    schema_store = {
+        data_schema['$id']: data_schema,
+        schema['$id']: schema,
+        definition_schema['$id']: definition_schema
+    }
+    content = request.get_json()
+
+    # Check if request is for a chart
+    if 'highChartsData' not in content:
+        logging.info("Not a charts request. Skipping...")
```

Comment:
We may accept other sources of charts data in the future, please specify highcharts
```suggestion
        logging.info("Not a highcharts request. Skipping...")
```